bindings/capacitorjs/ios/Plugin/Plugin.swift

```diff
@@ -0,0 +1,207 @@
+    private let error_buffer: UnsafeMutablePointer<CChar>? = nil
+    private let error_buffer_size = 0
+    
+    @objc func messageHandlerNew(_ call: CAPPluginCall) {
+        do {
+            print("Capacitor call messageHandlerNew received", call.jsObjectRepresentation)
+            guard let storagePath = call.getString("storagePath"),
+                  let clientOptions = call.getObject("clientOptions"),
+                  let coinType = call.getInt("coinType"),
+                  let secretManager = call.getObject("secretManager") else {
+                return call.reject("storagePath, clientOptions, coinType, and secretManager are required")
+            }
+            guard JSONSerialization.isValidJSONObject(clientOptions),
+                  JSONSerialization.isValidJSONObject(secretManager) else {
+                return call.reject("clientOptions or secretManager is an invalid JSON object")
+            }
+            let ClientOptions = try? JSONSerialization.data(withJSONObject: clientOptions)
+            // TODO: replacing for urls slashes temporaly, make better using Codable structs with URL type?
+            let stringfiedClientOptions = String(data: ClientOptions!, encoding: .utf8)!.replacingOccurrences(of: "\\", with: "")
+            
+            let SecretManager = try? JSONSerialization.data(withJSONObject: secretManager)
+            let stringfiedSecretManager = String(data: SecretManager!, encoding: .utf8)!
+            print(stringfiedSecretManager)
+            // prepare the internal app directory path
+            let fm = FileManager.default
+            guard let documents = fm.urls(
+                for: .applicationSupportDirectory,
+                   in: .userDomainMask
+            ).first else { 
+                return  call.reject("can not create the path")
+            }
+            let path = documents.appendingPathComponent(
+                storagePath,
+                isDirectory: true
+            ).path
+            
+            if !fm.fileExists(atPath: path) {
+                try fm.createDirectory(
+                    atPath: path,
+                    withIntermediateDirectories: true,
+                    attributes: nil
+                )
+            }
+            
+            // Exclude folder from auto-backup
+            var urlPath = URL(fileURLWithPath: path, isDirectory: true)
+            var values = URLResourceValues()
+            values.isExcludedFromBackup = true
+            try urlPath.setResourceValues(values)
+
+            // we need to modify the path on the JS object
+            let options = """
+            {
+                "storagePath":"\(path)",
+                "clientOptions":\(String(describing: stringfiedClientOptions)),
+                "coinType":\(coinType),
+                "secretManager":{
+                    "stronghold":{
+                        "snapshotPath":"\(path)/wallet.stronghold"
+                }}
```

Comment:
What happens if the secretManager is not stronghold?

Reply:
That's needs to be implemented yet. I've just added a new task here https://github.com/iotaledger/wallet.rs/issues/1959